### PR TITLE
HOSTEDCP-2070: add AKS e2e certificate testing for Hypershift

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
ci/rehearse/openshift/hypershift/main/e2e-aks

**Review by bryan-cox — COMMENTED** [review]
Just a question and a really small nit

File: ci-operator/step-registry/aks/grant-kv-access/aks-grant-kv-access-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+ref:
+  as: aks-grant-kv-access
+  from_image:
+    namespace: ocp
+    name: "4.16"
```
[the remark this answers is not in the document]
Comment (Patryk-Stefanski) [reply ci-operator/step-registry/aks/grant-kv-access/aks-grant-kv-access-ref.yaml:5]: Im not sure this part was cherry-picked from @fxierh work. @fxierh any reason not to use 4.17 ?

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

File: ci-operator/step-registry/aks/grant-kv-access/aks-grant-kv-access-ref.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fxierh) [reply ci-operator/step-registry/aks/grant-kv-access/aks-grant-kv-access-ref.yaml]: Here upi-installer is used as a provider for various CLIs tools so the specific version isn’t particularly relevant. Anyways I'll probably just remove this step and merge its content into aks-provision.

**Comment by fxierh** [comment]
Looking good.

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by bryan-cox** [comment]
```
{Failed  === RUN   TestNodePool/HostedCluster0/ValidateHostedCluster/EnsureNoCrashingPods
    util.go:526: Container attacher-kube-rbac-proxy in pod azure-disk-csi-driver-controller-68d785558-psz7l has a restartCount > 0 (1)
```
Looks like a flake

**Comment by bryan-cox** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Comment by Patryk-Stefanski** [comment]
/pj-rehearse pull-ci-openshift-hypershift-release-4.17-e2e-aks

**Pull request merged**